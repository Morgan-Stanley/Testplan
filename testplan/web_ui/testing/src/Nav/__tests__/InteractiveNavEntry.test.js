/* Unit tests for the InteractiveNavEntry component. */
import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

import InteractiveNavEntry from '../InteractiveNavEntry.js';
import {FakeInteractiveReport} from '../../Common/sampleReports.js';

describe('InteractiveNavEntry', () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders a testcase in "ready" state', () => {
    const renderedEntry = shallow(
      <InteractiveNavEntry
        name={'FakeTestcase'}
        status={'ready'}
        type={'testcase'}
        caseCountPassed={0}
        caseCountFailed={0}
        handlePlayClick={() => undefined}
      />
    );
    expect(renderedEntry).toMatchSnapshot();
  });

  it('renders a testcase in "running" state', () => {
    const renderedEntry = shallow(
      <InteractiveNavEntry
        name={'FakeTestcase'}
        status={'running'}
        type={'testcase'}
        caseCountPassed={0}
        caseCountFailed={0}
        handlePlayClick={() => undefined}
      />
    );
    expect(renderedEntry).toMatchSnapshot();
  });

  it('renders a testcase in "passed" state', () => {
    const renderedEntry = shallow(
      <InteractiveNavEntry
        name={'FakeTestcase'}
        status={'passed'}
        type={'testcase'}
        caseCountPassed={9}
        caseCountFailed={0}
        handlePlayClick={() => undefined}
      />
    );
    expect(renderedEntry).toMatchSnapshot();
  });

  it('renders a testcase in "failed" state', () => {
    const renderedEntry = shallow(
      <InteractiveNavEntry
        name={'FakeTestcase'}
        status={'failed'}
        type={'testcase'}
        caseCountPassed={8}
        caseCountFailed={1}
        handlePlayClick={() => undefined}
      />
    );
    expect(renderedEntry).toMatchSnapshot();
  });

  it('calls handlePlayClick when the play button is clicked', () => {
    const handlePlayClick = jest.fn();
    const renderedEntry = shallow(
      <InteractiveNavEntry
        name={'FakeTestcase'}
        status={'ready'}
        type={'testcase'}
        caseCountPassed={0}
        caseCountFailed={0}
        handlePlayClick={handlePlayClick}
      />
    );

    renderedEntry.find(FontAwesomeIcon).simulate('click');
    expect(handlePlayClick.mock.calls.length).toEqual(1);
  });

  it('calls handlePlayClick when the replay button is clicked', () => {
    const handlePlayClick = jest.fn();
    const renderedEntry = shallow(
      <InteractiveNavEntry
        name={'FakeTestcase'}
        status={'failed'}
        type={'testcase'}
        caseCountPassed={6}
        caseCountFailed={2}
        handlePlayClick={handlePlayClick}
      />
    );

    renderedEntry.find(FontAwesomeIcon).simulate('click');
    expect(handlePlayClick.mock.calls.length).toEqual(1);
  });
});

