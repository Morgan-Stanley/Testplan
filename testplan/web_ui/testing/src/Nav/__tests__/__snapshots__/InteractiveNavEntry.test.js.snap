// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractiveNavEntry renders a testcase in "failed" state 1`] = `
<div
  className="d-flex justify-content-between"
>
  <div
    className="entryName_1w2kiva-o_O-failed_11es5k7"
  >
    FakeTestcase
  </div>
  <div
    className="entryIcons_dx1noq"
  >
    <i
      className="entryIcon_ull2p3"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        8
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        1
      </span>
    </i>
    <Badge
      className="entryIcon_ull2p3-o_O-failedBadge_1wefnsd"
      color="secondary"
      pill={true}
      tag="span"
      title="testcase"
    >
      C
    </Badge>
    <FontAwesomeIcon
      border={false}
      className="entryButton_yr2g96"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f01e",
            "M500.333 0h-47.411c-6.853 0-12.314 5.729-11.986 12.574l3.966 82.759C399.416 41.899 331.672 8 256.001 8 119.34 8 7.899 119.526 8 256.187 8.101 393.068 119.096 504 256 504c63.926 0 122.202-24.187 166.178-63.908 5.113-4.618 5.354-12.561.482-17.433l-33.971-33.971c-4.466-4.466-11.64-4.717-16.38-.543C341.308 415.448 300.606 432 256 432c-97.267 0-176-78.716-176-176 0-97.267 78.716-176 176-176 60.892 0 114.506 30.858 146.099 77.8l-101.525-4.865c-6.845-.328-12.574 5.133-12.574 11.986v47.411c0 6.627 5.373 12 12 12h200.333c6.627 0 12-5.373 12-12V12c0-6.627-5.373-12-12-12z",
          ],
          "iconName": "redo",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Run tests"
      transform={null}
    />
  </div>
</div>
`;

exports[`InteractiveNavEntry renders a testcase in "passed" state 1`] = `
<div
  className="d-flex justify-content-between"
>
  <div
    className="entryName_1w2kiva-o_O-passed_1kh4m98"
  >
    FakeTestcase
  </div>
  <div
    className="entryIcons_dx1noq"
  >
    <i
      className="entryIcon_ull2p3"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        9
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        0
      </span>
    </i>
    <Badge
      className="entryIcon_ull2p3-o_O-passedBadge_1gkx146"
      color="secondary"
      pill={true}
      tag="span"
      title="testcase"
    >
      C
    </Badge>
    <FontAwesomeIcon
      border={false}
      className="entryButton_yr2g96"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f01e",
            "M500.333 0h-47.411c-6.853 0-12.314 5.729-11.986 12.574l3.966 82.759C399.416 41.899 331.672 8 256.001 8 119.34 8 7.899 119.526 8 256.187 8.101 393.068 119.096 504 256 504c63.926 0 122.202-24.187 166.178-63.908 5.113-4.618 5.354-12.561.482-17.433l-33.971-33.971c-4.466-4.466-11.64-4.717-16.38-.543C341.308 415.448 300.606 432 256 432c-97.267 0-176-78.716-176-176 0-97.267 78.716-176 176-176 60.892 0 114.506 30.858 146.099 77.8l-101.525-4.865c-6.845-.328-12.574 5.133-12.574 11.986v47.411c0 6.627 5.373 12 12 12h200.333c6.627 0 12-5.373 12-12V12c0-6.627-5.373-12-12-12z",
          ],
          "iconName": "redo",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Run tests"
      transform={null}
    />
  </div>
</div>
`;

exports[`InteractiveNavEntry renders a testcase in "ready" state 1`] = `
<div
  className="d-flex justify-content-between"
>
  <div
    className="entryName_1w2kiva"
  >
    FakeTestcase
  </div>
  <div
    className="entryIcons_dx1noq"
  >
    <i
      className="entryIcon_ull2p3"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        0
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        0
      </span>
    </i>
    <Badge
      className="entryIcon_ull2p3"
      color="secondary"
      pill={true}
      tag="span"
      title="testcase"
    >
      C
    </Badge>
    <FontAwesomeIcon
      border={false}
      className="entryButton_yr2g96"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f04b",
            "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
          ],
          "iconName": "play",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Run tests"
      transform={null}
    />
  </div>
</div>
`;

exports[`InteractiveNavEntry renders a testcase in "running" state 1`] = `
<div
  className="d-flex justify-content-between"
>
  <div
    className="entryName_1w2kiva"
  >
    FakeTestcase
  </div>
  <div
    className="entryIcons_dx1noq"
  >
    <i
      className="entryIcon_ull2p3"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        0
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        0
      </span>
    </i>
    <Badge
      className="entryIcon_ull2p3"
      color="secondary"
      pill={true}
      tag="span"
      title="testcase"
    >
      C
    </Badge>
    <Loader
      color="#123abc"
      css={Object {}}
      height={4}
      heightUnit="px"
      loading={true}
      size={4}
      width={100}
      widthUnit="px"
    />
  </div>
</div>
`;
